src/platform/packages/shared/kbn-esql-utils/src/utils/run_query.ts

```diff
@@ -38,6 +39,24 @@ export const getStartEndParams = (query: string, time?: TimeRange) => {
   return [];
 };
 
+export const getNamedParams = (
+  query: string,
+  timeRange?: TimeRange,
+  variables?: ESQLControlVariable[]
+) => {
+  const namedParams: ESQLSearchParams['params'] = getStartEndParams(query, timeRange);
+  if (variables?.length) {
+    variables?.forEach(({ key, value, type }) => {
+      if (type === ESQLVariableType.FIELDS) {
```

Comment:
ℹ️ Super important note: ES needs a different payload if the variable value is a field or function.